add qnn test template job

cccclai opened the pull request:
Differential Revision: D68112936

haowhsu-quic commented on .github/workflows/pull.yml:
```diff
@@ -395,6 +395,25 @@ jobs:
         PYTHON_EXECUTABLE=python bash .ci/scripts/test_llama.sh -model stories110M -build_tool "${BUILD_TOOL}" -mode "${MODE}" -dtype "${DTYPE}" -pt2e_quantize "${PT2E_QUANTIZE}"
 
+  test-qnn-models-linux:
+    name: test-mediatek-models-linux
```
Might be `qnn` I think.

haowhsu-quic commented:
Thank you for this, we 'll figure out which test cases are mandatory. I think we could just have them run on the emulator, or you have some connected hardware could be used?

cccclai commented:
> Thank you for this, we 'll figure out which test cases are mandatory. I think we could just have them run on the emulator, or you have some connected hardware could be used?

We can only run emulators on the CI for now. However, @guangy10 is working on the perf benchmark with real device and he has enabled to qnn model running there. I notice that there are some real models in the test_qnn_delegate.py and we can have those running on real device. They will run periodically, but not per pull request, though should be good enough to catch failure and track down the issue.

cccclai committed:
add qnn test template job (#7636)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/7636

Differential Revision: D68112936

huydhn commented on .github/workflows/pull.yml:
```diff
@@ -395,6 +395,25 @@ jobs:
+      docker-image: executorch-ubuntu-22.04-qnn-sdk
+      submodules: 'true'
+      ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
+      timeout: 90
```
I increase the timeout value to a large number because of a recent change to wait for the Docker image to be available, which could take up to an hour when it happens

```suggestion
      timeout: 180
```

huydhn committed:
Increase timeout

zingo commented:
Hi
This might have broken pull and trunk jobs since around this patch got merged they are crashing , this PR also seem to have the same problem:  
https://hud.pytorch.org/pr/pytorch/executorch/7636

zingo commented:
Might not be this patch, I tried a "revert" PR and get the same issue in it so it might probably be something else causing it.

huydhn commented:
Ops, that is a real error, let me submit a forward fix

huydhn commented:
The fix PR https://github.com/pytorch/executorch/pull/7731

guangy10 commented:
@cccclai We already have the qnn models in trunk.yml: https://github.com/pytorch/executorch/blob/main/.github/workflows/trunk.yml#L302-L323, which will be triggered on PR if files under tracked paths got modified. Why do we need to add same to pull.yml?

cccclai commented:
> @cccclai We already have the qnn models in trunk.yml: https://github.com/pytorch/executorch/blob/main/.github/workflows/trunk.yml#L302-L323, which will be triggered on PR if files under tracked paths got modified. Why do we need to add same to pull.yml?

oh this pr is to cover the perf op test defined in https://github.com/pytorch/executorch/blob/57a09f49027ab05515afef4315d0a797657222f3/backends/qualcomm/tests/test_qnn_delegate.py#L4
